File: selectionSort/selectionSort.js
function selectionSort(unsortedArray)
{
    let n = unsortedArray.length;
    for(let i = 0; i< n-1; i++)
    {
        let index = i;
        for(let j = i+1; j<n; j++)
        {
            if(unsortedArray[j] < unsortedArray[index])
            {
                index = j;
            }
        }
        let temp = unsortedArray[i];
        unsortedArray[i] = unsortedArray[index];
        unsortedArray[index] = temp;
    }
}

File: selectionSort/selectionSort.cpp
#include <vector>
#include <iostream>

std::vector<int> selectionSort(std::vector<int> unsortedArray)
{
    int n = unsortedArray.size();
    for (int i = 0; i < n; i++)
    {
        int index = i;
        for (int j = i; j < n; j++)
        {
            if (unsortedArray[index] > unsortedArray[j])
            {
                index = j;
            }
        }
        int temp = unsortedArray[i];
        unsortedArray[i] = unsortedArray[index];
        unsortedArray[index] = temp;
    }

    return unsortedArray;
}